### PR TITLE
Enhancement/update metadata v2 w schemas

### PR DESCRIPTION
**Pull request by w-aldrich**

### CODE REVIEW

File: ingestors/kafka/src/main/scala/hydra/kafka/model/TopicMetadataV2Transport.scala [hunk 1]
```diff
@@ -52,9 +53,16 @@ object ContactMethod {
   }
 }
 
+object Schemas {
+  def emptySchema: Schema = SchemaBuilder
```
Comment (lewisjkl) [inline ingestors/kafka/src/main/scala/hydra/kafka/model/TopicMetadataV2Transport.scala:57]: When would this function actually be used? Is this just in sending data back to the front-end if the schema doesn't exist?

Comment (w-aldrich) [reply ingestors/kafka/src/main/scala/hydra/kafka/model/TopicMetadataV2Transport.scala:57]: That is exactly right.

File: ingestors/kafka/src/test/scala/hydra/kafka/serializers/TopicMetadataV2ParserSpec.scala [hunk 2]
```diff
@@ -428,8 +428,8 @@ class TopicMetadataV2ParserSpec extends AnyWordSpecLike with Matchers {
         parentSubjects = parentSubjects,
         notes = notes
       )
-
-      TopicMetadataV2Format.write(topicMetadataV2) shouldBe
+      val a = TopicMetadataV2Format.write(topicMetadataV2)
```
Comment (lewisjkl) [inline ingestors/kafka/src/test/scala/hydra/kafka/serializers/TopicMetadataV2ParserSpec.scala:431]: Guessing this `a` was for debugging

File: ingestors/kafka/src/main/scala/hydra/kafka/model/TopicMetadataV2Transport.scala (hunk 1 above)
Comment (lewisjkl) [reply ingestors/kafka/src/main/scala/hydra/kafka/model/TopicMetadataV2Transport.scala:57]: Would it make more sense to make the field optional in the metadata when we send it down to the front end?

Comment (bretrbs) [reply ingestors/kafka/src/main/scala/hydra/kafka/model/TopicMetadataV2Transport.scala:57]: Just explodes issues throughout the project. We felt like this solution minimizes that. The change can be made, we just need to decide what to do with optional schemas throughout the project where they really can't be optional (it's a parse error if they don't provide it on the way in).

Comment (lewisjkl) [reply ingestors/kafka/src/main/scala/hydra/kafka/model/TopicMetadataV2Transport.scala:57]: Makes sense. I think that I worry about what the `emptySchema` will cause to happen throughout the project. Could be a crutch that allows us to ignore error cases that are important.

If the schema has to be there, and truly isn't optional, then we should have it show up like the error that it is. If we read in a piece of metadata that does not have an associated schema, we should probably just not add that metadata to the cache and log out a `WARN`. If we wanted to, we could even keep a separate list of metadata or topic names that have no schema.. idk Thoughts?

**Pull request merged**